lib/pure/os.nim

```diff
@@ -281,18 +354,53 @@ proc tailDir*(path: string): string {.
 
 proc isRootDir*(path: string): bool {.
   noSideEffect, rtl, extern: "nos$1".} =
-  ## Checks whether a given `path` is a root directory
+  ## Checks whether a given `path` is a root directory.
+  runnableExamples:
+    assert isRootDir("")
+    assert isRootDir(".")
+    assert isRootDir("/")
+    assert isRootDir("a")
+    assert not isRootDir("/a")
+    assert not isRootDir("a/b/c")
+
   result = parentDirPos(path) < 0
 
 iterator parentDirs*(path: string, fromRoot=false, inclusive=true): string =
-  ## Walks over all parent directories of a given `path`
+  ## Walks over all parent directories of a given `path`.
```

Comment:
we really shd fix https://github.com/nim-lang/Nim/issues/8353 so that this description remains valid

Reply:
From what I have seen, you have created #8177 with lots of issues regarding `os.nim`.

I think that is orthogonal to this PR, i.e. if/when some of those get fixed/changed, the documentation should be changed accordingly.

Reply:
sure, it's indeed orthogonal.